mlflow/genai/scorers/builtin_scorers.py

```diff
@@ -1564,6 +1571,121 @@ def __call__(
+
+    You can invoke the scorer directly with a session for testing, or pass it to
+    `mlflow.genai.evaluate` for running full evaluation on a dataset.
+
+    Args:
+        name: The name of the scorer. Defaults to "user_frustration".
+        model: {{ model }}
+
+    Example (direct usage):
+
+    .. code-block:: python
+
+        import mlflow
+        from mlflow.genai.scorers import UserFrustration
+
+        # Retrieve a list of traces with the same session ID
+        session = traces_from_my_session = mlflow.search_traces(
+            experiment_ids=[experiment_id],
+            filter_string=f"metadata.`mlflow.trace.session` = '{session_id}'",
+            return_type="list",
+        )
+
+        assessment = UserFrustration(name="my_user_frustration_judge")(session=session)
+        print(assessment)  # Feedback with value "frustrated" or "not_frustrated"
+
+    Example (with evaluate):
+
+    .. code-block:: python
+
+        import mlflow
+        from mlflow.genai.scorers import UserFrustration
+
+        session = traces_from_my_session = mlflow.search_traces(
+            experiment_ids=[experiment_id],
+            filter_string=f"metadata.`mlflow.trace.session` = '{session_id}'",
+            return_type="list",
+        )
+        result = mlflow.genai.evaluate(data=session, scorers=[UserFrustration()])
+    """
+
+    name: str = USER_FRUSTRATION_ASSESSMENT_NAME
+    model: str | None = None
+    required_columns: set[str] = {"session"}
+    description: str = "Evaluate whether the agent's response causes user frustration."
+    _judge: InstructionsJudge | None = pydantic.PrivateAttr(default=None)
+
+    def _get_judge(self) -> InstructionsJudge:
+        """Return a session-level InstructionsJudge with the user frustration prompt."""
+        if self._judge is None:
+            self._judge = InstructionsJudge(
+                name=self.name,
+                instructions=self.instructions,
+                model=self.model,
+                description=self.description,
+                feedback_value_type=Literal["frustrated", "not_frustrated"],
+            )
+        return self._judge
+
+    @property
+    def is_session_level_scorer(self) -> bool:
+        """Return True as UserFrustration requires a session of traces for evaluation."""
+        return True
+
+    @property
+    def instructions(self) -> str:
+        """Get the instructions of what this scorer evaluates."""
+        return USER_FRUSTRATION_PROMPT
+
+    def get_input_fields(self) -> list[JudgeField]:
+        """
+        Get the input fields for the UserFrustration judge.
+
+        Returns:
+            List of JudgeField objects defining the input fields based on the __call__ method.
+        """
+        return [
+            JudgeField(
+                name="session",
+                description="A list of trace objects belonging to the same conversation session.",
+            ),
+        ]
```

Comment:
WDYT of adding a child class of BuiltInScorer, BuiltInSessionLevelScorer. It would have implementations for `get_input_fields`, `is_session_level_scorer`, and `__call__`.

Reply:
I think it's a reasonable assumption to make. Also think this would be good to have. `__call__` seems standard enough to have in the `BuiltInSessionLevelScorer` as well